docs: document mixing named and default slots (#1946)
vuejs/docs: src/guide/components/slots.md

Note the `name` of a slot won't be included in the props because it is reserved - so the resulting `headerProps` would be `{ message: 'hello' }`.

> Also note that if you are mixing named scoped slots with the default scoped slot, you need to explicitly define the default template. Otherwise, it would hint that the data of the default scoped slot would be available in the other slots scopes.

Reviewer: ```suggestion
If you are mixing named slots with the default scoped slot, you need to use an explicit `<template>` tag for the default slot. Attempting to place the `v-slot` directive directly on the component will result in a compilation error. This is to avoid any ambiguity about the scope of the props of the default slot. For example:
```

> ```vue-html

Reviewer: It's always a bit dangerous to have examples of incorrect code. People tend to skim over the text and just copy the examples. I think a note at the top of the example can help to mitigate that.

```suggestion
```vue-html
<!-- This template won't compile -->
```

<template>

>   <BaseLayout v-slot="{ message }">
>     <p>{{ message }}</p>
>     <template #footer>
>       <!-- message belongs to the default slot, and is not available here -->
>       <p>{{ message }}</p>
>     </template>
>   </BaseLayout>

Reviewer: I think the two examples should use the same component name. As the preceding example also uses `MyComponent` I decided to go with that, but we could make them both `BaseLayout` if you prefer.

```suggestion
  <MyComponent v-slot="{ message }">
    <p>{{ message }}</p>
    <template #footer>
      <!-- message belongs to the default slot, and is not available here -->
      <p>{{ message }}</p>
    </template>
  </MyComponent>
```

Author: I'm sorry, it was clearly a failed copy paste from my local example, it's normal to have `MyComponent`

</template>
```

> Instead, you will need to write it as follows:

Reviewer: ```suggestion
Using an explicit `<template>` tag for the default slot helps to make it clear that the `message` prop is not available inside the other slot:

 ```
